DARPA-ASKEM/terarium
# feat: Kernel message dispatcher

### Summary
Adding a wrapper around `services/jupyter` to do context setup, and to automatically route messages to the correct senders.

Because messaging is not 1-to-1, a request can result in multiple, asynchronous responses coming back. I added a manager class to handle the message routing based on Event-emitter and unique-id embedded into request/response messages.

Note there is kind of a busy-wait mechanism for setting up the context, as this doesn't happen synchronously either.


### Testing
There is no UI. 
- Open debugger console
- Navigate to http://localhost:8080/jupyter-test
- In the console you should be able to see something like this, in order
```
waiting... KernelConnection {_id: '073cdb4f-9e78-44dc-9434-69cc9178cf73', _createSocket: ƒ, _onWSOpen: ƒ, _onWSMessage: ƒ, _onWSClose: ƒ, …}

stratify_response {header: {…}, msg_id: 'c8d39ffe-bf3e-4f47-8ec4-cf489b70dfc0', msg_type: 'stratify_response', parent_header: {…}, metadata: {…}, …}
```
